### PR TITLE
Laz-26: Creating trail with position time to live

### PR DESCRIPTION
**Pull request by m4rtint**
Demo
- https://perigongames.slack.com/archives/C01NGV9PN3D/p1615656685004400

Created
- Created SandboxLazo.unity 
- LazoPosition struct
- Added prototype texture floor
- ScriptableObjects  for LazoProperties- Time To Live and positions
- Created Mock Movement file

Changes
- Used Linq to decrement time per position, and filter out the position that are dead
- Moved files into correct folders

### CODE REVIEW

File: Assets/Scripts/Lazo.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            _listOfPositions.Clear();
+        }
+
+        private void ResetRateOfRecordingTimeElapsed()
+        {
+            _rateOfRecordingTimerElapsed = _lazoProperties.RateOfRecordingPosition;
+        }
+        #endregion
+
+        #region Debug
```
Comment (m4rtint) [inline Assets/Scripts/Lazo.cs:98]: This Debug puts down a cube for each recorded position

File: Assets/Scripts/Lazo.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                RunLazo(position);
+                ResetRateOfRecordingTimeElapsed();
+            }
+
+            RemoveOldestPointIfNeeded(deltaTime);
+        }
+
+        private void RemoveOldestPointIfNeeded(float deltaTime)
+        {
+            _listOfPositions = _listOfPositions.Select(lazoPosition =>
```
Comment (m4rtint) [inline Assets/Scripts/Lazo.cs:53]: You can imagine this section as a mapping function
- meaning - for each LazoPosition...do all this underneath.

File: Assets/Scripts/Lazo.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            {
+                lazoPosition.TimeToLive -= deltaTime;
+                // Debug
+                if (lazoPosition.TimeToLive < 0)
+                {
+                    DebugDestroyLastCubeOnList();
+                }
+                return lazoPosition;
+            }).Where(lazoPosition => lazoPosition.TimeToLive > 0).ToList();
```
Comment (m4rtint) [inline Assets/Scripts/Lazo.cs:62]: Where - it's like a filter function. 
(Only keep lazoPosition where the timeToLive is still > 0

File: Assets/Scripts/Lazo.cs [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        #region Helper
+        private void RunLazo(Vector3 position)
+        {
+            if (_listOfPositions.Count > 0 &&
+                _listOfPositions.Last().Position == position)
+            {
+                return;
+            }
+
+            Debug.Log("Number of Position: " + _listOfPositions.Count);
```
Comment (m4rtint) [inline Assets/Scripts/Lazo.cs:75]: Thinking of keeping this for when we build more on Lazo

File: Assets/Scripts/Lazo.cs [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        #endregion
+
+        #region Debug
+        /*
+         * DEBUG
+         */
+        private List<GameObject> _listOfDebugCubes = new List<GameObject>();
+        private void DebugCreateCubeAt(Vector3 position)
+        {
+            Debug.Log("Number of Cube: "+ _listOfDebugCubes.Count);
```
Comment (m4rtint) [inline Assets/Scripts/Lazo.cs:105]: Debug

**Review by wbeltran777 — APPROVED** [review]
Sweet, this looks great. I like the filter function for applying the time delta on each point.

**Pull request merged**